src/pint/models/model_builder.py

```diff
@@ -24,295 +25,402 @@
+        timing_model: `pint.models.TimeModel` object
+            Timing model to add the parameters to.
+        params: list
+            A list of number unknown parameter names.
 
+        Note
+        ----
+        This function do not fill the parameter values. Only adds the bare
+        parameter to the timing model.
+        """
+        # go over the input parameter list
+        unknown_param = []
+        for pp in indexed_params:
+            pint_p = self.alias_2_pint_param(pp)
+            # A true unrecognized name
+            if pint_p is None:
+                unknown_param.append(pp)
+                continue
+            # Check if parameter in the timing model already
+            if pint_p in timing_model.params:
+                continue
+            # Check if parameter name has and index and prefix
+            try:
+                prefix, idx_str, idx = split_prefixed_name(pint_p)
+            except PrefixError:
+                prefix = None
+                idx = None
+            if prefix: #
+                search_name = prefix
+            else:
+                search_name = pint_p
+            # TODO, when the prefix parameter structure changed, this will have
+            # to change.
+            prefix_map = timing_model.get_prefix_mapping(search_name)
+            if prefix_map == {}: # Can not find any prefix mapping
+                unknown_param.append(pp)
+                continue
+            # Get the parameter in the prefix map.
+            prefix_param0 = list(prefix_map.items())[0]
+            example = getattr(timing_model, prefix_param0[1])
+            if not idx: # Input name has index, init from an example param and add it to timing model.
+                idx = max(list(prefix_map.keys())) + 1
+            host_component = timing_model._locate_param_host(prefix_param0[1])[0][0]
+            timing_model.add_param_from_top(example.new_param(idx), host_component)
+        return timing_model, unknown_param
+
+    def _report_conflict(self, conflict_graph):
+        """ Report conflict components
+        """
+        for k, v in conflict_graph.items():
+            # Put all the conflict components together from the graph
+            cf_cps = v.append(k)
+            raise ComponentConflict("Can not decide the one component from:"
+                                    " {}".format(cf_cps))
 
 def get_model(parfile):
     """A one step function to build model from a parfile
-
     Parameters
     ----------
     parfile : str
         The parfile name, or a file-like object to read the parfile contents from
-
     Returns
     -------
     Model instance get from parfile.
-
     """
     try:
         contents = parfile.read()
     except AttributeError:
         contents = None
     if contents is None:
         # parfile is a filename and can be handled by ModelBuilder
-        mbmodel = ModelBuilder(parfile)
-        model = mbmodel.timing_model
+        mbmodel = ModelBuilder()
```

Comment:
Shouldn't there be one ModelBuilder object created for the entire program? This will make the lazyproperties work. Just set `model_builder = None` at the module level, then check if it's None and create one to replace it if necessary here.